SimPEG/ObjectiveFunction.py

```diff
@@ -371,6 +400,64 @@ def W(self):
                 W.append(curW)
         return sp.vstack(W)
 
+    def expose(self, properties):
```

Comment:
I would appreciate a quick review on this by someone well-versed in python (@rowanc1, @bsmithyman, @jcapriot :) ). In particular - is messing with `__getattribute__` and `__setattr__` as in lines 444 a terrible idea??